framework/src/csg/CSGCellList.C

```diff
@@ -0,0 +1,94 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CSGCellList.h"
+
+namespace CSG
+{
+
+CSGCellList::CSGCellList() {}
+
+void
+CSGCellList::checkCellName(const std::string & name) const
+{
+  if (_cells.find(name) != _cells.end())
+    mooseError("Cell with name " + name + " already exists in geometry.");
+}
```

Comment:
```suggestion
CSGCell &
CSGCellList::addCell(std::unique_ptr<CSGCell> cell)
{
  auto [it, inserted] = _cells.emplace(cell->name(), std::move(cell));
  if (!inserted)
    mooseError("Cell with name " + name + " already exists in geometry.");
  return it->second;
}
```

and make this a protected method. This will replace the other `addCell` method.

The emplace call returns a pair of the iterator to the inserted element (or current element) and a bool as to if the insertion took place. So checking second here (if false) means that something already existed.

And then in the add cell routines below, just do something like (`addVoidCell` example):

```
CSGCell &
CSGCellList::addVoidCell(const std::string & name, const CSGRegion & region)
{
  return addCell(name, std::make_unique<CSGCell>(name, region));
}
```

This changes three lookups (if it exists, on insertion, and getting it) one